### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by allycattiny** [FossilsArcheologyRevival/FossilsArcheologyRevival#1324]
Hello, I am allycattiny on twitter and I got directed here after issuing a complaint through twitter. I am running the 1.12 version on the mod and it is really unstable. it barely runs and crashes every time it has to update any chunks at all. For clarification, I have used this mod since it was for minecraft 1.2 (before it was revival) so i doubt it installed incorrectly. I am running it alongside Optifine and with the default texture pack. The only thing I can think of is that since I was running a mod pack that I had made for 1.7 before I downloaded this version that maybe I didnt correctly remove all of the mods/configs but I doubt that would cause these issues (it is also only allocated 4gb of ram if that's relevant). Attached is the crash report of the most recent crash.

---- Minecraft Crash Report ----

WARNING: coremods are present:
  llibrary (llibrary-core-1.0.10-1.12.2.jar)
Contact their authors BEFORE contacting forge

// Daisy, daisy...

Time: 4/15/19 12:18 AM
Description: Exception ticking world

java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1429)
	at java.util.HashMap$ValueIterator.next(HashMap.java:1458)
	at net.minecraft.world.chunk.Chunk.func_76623_d(Chunk.java:870)
	at net.minecraft.world.gen.ChunkProviderServer.func_73156_b(ChunkProviderServer.java:269)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:207)
	at net.optifine.override.WorldServerOF.func_72835_b(WorldServerOF.java:39)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:252)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Thread.java:745)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1429)
	at java.util.HashMap$ValueIterator.next(HashMap.java:1458)
	at net.minecraft.world.chunk.Chunk.func_76623_d(Chunk.java:870)
	at net.minecraft.world.gen.ChunkProviderServer.func_73156_b(ChunkProviderServer.java:269)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:207)
	at net.optifine.override.WorldServerOF.func_72835_b(WorldServerOF.java:39)

-- Affected level --
Details:
	Level name: New Dinosaur World 2
	All players: 1 total; [EntityPlayerMP['allycat247'/343, l='New Dinosaur World 2', x=500.18, y=84.48, z=379.74]]
	Chunk stats: ServerChunkCache: 550 Drop: 4
	Level seed: 960570313
	Level generator: ID 00 - default, ver 1. Features enabled: true
	Level generator options: 
	Level spawn location: World: (240,64,232), Chunk: (at 0,4,8 in 15,14; contains blocks 240,0,224 to 255,255,239), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 300077 game time, 175894 day time
	Level dimension: 0
	Level storage version: 0x04ABD - Anvil
	Level weather: Rain time: 19377 (now: false), thunder time: 874 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: true
Stacktrace:
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:252)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Thread.java:745)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 333226112 bytes (317 MB) / 872415232 bytes (832 MB) up to 4294967296 bytes (4096 MB)
	JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx4G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	IntCache: cache: 0, tcache: 0, allocated: 12, tallocated: 94
	FML: MCP 9.42 Powered by Forge 14.23.5.2833 Optifine OptiFine_1.12.2_HD_U_E3 6 mods loaded, 6 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State  | ID        | Version      | Source                          | Signature                                |
	|:------ |:--------- |:------------ |:------------------------------- |:---------------------------------------- |
	| LCHIJA | minecraft | 1.12.2       | minecraft.jar                   | None                                     |
	| LCHIJA | mcp       | 9.42         | minecraft.jar                   | None                                     |
	| LCHIJA | FML       | 8.0.99.99    | forge-1.12.2-14.23.5.2833.jar   | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| LCHIJA | forge     | 14.23.5.2833 | forge-1.12.2-14.23.5.2833.jar   | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| LCHIJA | llibrary  | 1.7.17       | llibrary-1.7.17-1.12.2 (1).jar  | b9f30a813bee3b9dd5652c460310cfcd54f6b7ec |
	| LCHIJA | fossil    | 8.0.0        | fossilsarcheology-8.0.0 (1).jar | None                                     |

	Loaded coremods (and transformers): 
llibrary (llibrary-core-1.0.10-1.12.2.jar)
  net.ilexiconn.llibrary.server.core.plugin.LLibraryTransformer
  net.ilexiconn.llibrary.server.core.patcher.LLibraryRuntimePatcher
	GL info: ~~ERROR~~ RuntimeException: No OpenGL context found in the current thread.
	Profiler Position: N/A (disabled)
	Player Count: 1 / 8; [EntityPlayerMP['allycat247'/343, l='New Dinosaur World 2', x=500.18, y=84.48, z=379.74]]
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'fml,forge'
	OptiFine Version: OptiFine_1.12.2_HD_U_E3
	OptiFine Build: 20181210-121000
	Render Distance Chunks: 8
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.3.0 - Build 20.19.15.4549
	OpenGlRenderer: Intel(R) HD Graphics 4400
	OpenGlVendor: Intel
	CpuCount: 4

**Comment by 4f6f3b** [comment]
Does this happen without Optifine? Seems like something isn't playing nice with it. Are you using default Optifine settings?

**Comment by allycattiny** [comment]
It crashes more without optifine unfortunately. That's why I messaged you.
Messing with some optifine settings has stabilised it slightly but it still
crashes.

**Comment by Canus8** [comment]
I might have made a separate thread for this, so could you tell what is causing the crash?  I had 2 mammoths who were causing immense lag, even down at diamond level (they were a little above default sea level), and the type of lag they were causing had crashed my game in the past, when naturally-spawning Jurassicraft brachiosaurus and velociraptors were causing the same kind of lag.  The Jurassicraft issue is related to an AI infinite loop, it seems.  

So, in short, could you identify if the crash was related to mammoths in some way?  If so, please delete my separate thread on the issue.

**Comment by lindstheginj** [comment]
Canus, PLEASE do not do this. There is no reason to bring your mammoth lag issue into a thread for a crash with no mention of mammoths.

**Comment by Canus8** [comment]
But there is a possibility they are related.  How am I supposed to know whether or not they are related?  If they are, then I would have made an un-needed thread all together.  

Forgive me for not being a complete expert in this system, or in programming in general.  I am only a freshman of mechanical engineering, and I just felt like trying to help inform some people of some things that they or may not have been aware of.  If I can't give insight of any kind, then tell me.  I will quit using github if you want that.  I am just new to this kind of thing.  

Sorry.

**Comment by lindstheginj** [comment]
Like I’ve said before, keep individual things separate. Unless you are having the same issue as someone else (and in this case, there are no similarities between ally’s crash, which seems to happen immediately upon loading, and your lag, which only happens around certain movs), your bug should be listed as its own thing. It is much more efficient for us to close accidental repeat issues than for us to have two issues on one thread, because we close individual issues when they are fixed like a checklist. If there are two issues in one thread, we may miss the second bug when fixing.

**Comment by Sunconure11** [comment]
I've run into this issue. It's been fixed in dev. Wait for the next build to drop.

**Comment by allycattiny** [comment]
Oh so it will be fixed? Thank you for your response!

**Comment by lindstheginj** [comment]
Just want to follow up on this since the issue is still open - @allycattiny are you having this issue still in 8.0.1?

**Comment by allycattiny** [comment]
It takes more but no it's still a problem.

I just and in creative and travelled as far as fast as I could.
Got further than I expected but no it still overticked and crashed.

**Comment by allycattiny** [comment]
@lindstheginj 
I've just loaded it up after crashing and now I'm stuck. I can't move without it crashing.

**Comment by Alex-the-666** [comment]
fixed in dev
